### PR TITLE
window for transcript (#323)

### PR DESCRIPTION
**Pull request by rivalez**

### CODE REVIEW

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java [hunk 1]
```diff
@@ -32,9 +30,8 @@ BoardFX getBoardFX() {
         return boardFX;
     }
 
-    Board(final boolean enemy, final Output output) {
+    Board(final boolean enemy) {
```
Comment (brrt2) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java]: boolean final in this case looks a little strange

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java [hunk 2]
```diff
@@ -130,11 +120,6 @@ public void start(final Stage primaryStage) {
 
     @Override
     public void run() {
-        try {
-            shipsPlaced.acquire();
-        } catch (InterruptedException e) {
-            output.send(e.getMessage());
-        }
         while (true) {
```
Comment (brrt2) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java]: while(true) - there should be an ending condition

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java [hunk 3]
```diff
@@ -19,16 +17,20 @@
...
-    private Semaphore shipsPlaced;
+    private Semaphore myTurn;
+    private boolean firstPlayer;
+    private Semaphore waitForSending;
     private boolean areAllShipsPlaced = false;
     private Queue<Integer> typesOfShips = new LinkedList<>(Arrays.asList(FOUR_MAST, THREE_MAST, THREE_MAST,
             DOUBLE_MAST, DOUBLE_MAST, DOUBLE_MAST, SINGLE_MAST, SINGLE_MAST, SINGLE_MAST, SINGLE_MAST));
 
-    ShipPlacer(Board enemyBoard, Board playerBoard, SocketServer socketServer, Semaphore shipsPlaced) {
+    ShipPlacer(Board enemyBoard, Board playerBoard, SocketServer socketServer, Semaphore myTurn, boolean firstPlayer, Semaphore waitForSending) {
```
Comment (brrt2) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java]: too many args in the constructor

File: battleship-server/src/main/java/com/academy/solid/nie/server/Game.java [hunk 4]
```diff
@@ -7,36 +7,47 @@
...
 
     Game(Player first, Player second) {
         this.first = first;
-        this.second = second;
         this.currentPlayer = first;
         this.otherPlayer = second;
     }
 
     void play() throws IOException {
+        log("It is now turn of " + getNameOfCurrentPlayer());
```
Comment (brrt2) [inline battleship-server/src/main/java/com/academy/solid/nie/server/Game.java]: String.format for the logs ?

File: battleship-server/src/main/java/com/academy/solid/nie/server/ServerGameInitializer.java [hunk 5]
```diff
@@ -49,8 +51,20 @@ private void exchangeShips() throws IOException {
         String secondShips = second.provideShips();
         first.inform(secondShips);
         second.inform(firstShips);
-
-        log("First's ships:" + firstShips + "\n" + "Second's ships:" + secondShips);
+        StringBuilder stringBuilder = new StringBuilder("Ships of First Player are located at:");
+        Arrays.stream(firstShips.split("\\|")).forEach(ship -> {
```
Comment (brrt2) [inline battleship-server/src/main/java/com/academy/solid/nie/server/ServerGameInitializer.java]: is the separator described somewhere ?

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/StartSceneController.java [hunk 6]
```diff
@@ -50,7 +50,8 @@
     void initialize() {
         Configuration configuration = new FileConfiguration();
         configuration.provide();
-        Language defaultLanguage = Language.valueOf(configuration.getCommunicate(ConfigProperty.LANGUAGE));
+        String language = configuration.getCommunicate(ConfigProperty.LANGUAGE).toUpperCase();
```
Comment (brrt2) [inline battleship-client/src/main/java/com/academy/solid/nie/client/ui/StartSceneController.java]: Please provide a validation or a default value

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java (hunk 1 above)
Comment (rivalez) [reply battleship-client/src/main/java/com/academy/solid/nie/client/ui/Board.java]: ok

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java (hunk 2 above)
Comment (rivalez) [reply battleship-client/src/main/java/com/academy/solid/nie/client/ui/GameScene.java]: ok

File: battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java (hunk 3 above)
Comment (rivalez) [reply battleship-client/src/main/java/com/academy/solid/nie/client/ui/ShipPlacer.java]: idea to separate it?

**Pull request merged**